src/java/htsjdk/samtools/util/AbstractProgressLogger.java

```diff
@@ -76,7 +76,7 @@ public synchronized boolean record(final String chrom, final int pos) {
      */
     @Override
     public synchronized boolean record(final SAMRecord rec) {
-        if (rec.getReferenceIndex() == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
+        if (null == rec.getHeader() || rec.getReferenceIndex() == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
             return record(null, 0);
```

Comment:
Recommend changing this method to use the reference name exclusively. Eg.,

    if ( SAMRecord.NO_ALIGNMENT_REFERENCE_NAME.equals(rec.getReferenceName()) ) {
        return record(null, 0);
    }
    else {
        return record(rec.getReferenceName(), rec.getAlignmentStart());
    }